### ISSUE TITLE
Consumes 100% of RAM when trying to identify a djvu file

### ISSUE THREAD

**Issue by glyh** [ImageMagick/ImageMagick#6584]
### ImageMagick version

git 2b4275854

### Operating system

Linux

### Operating system, version and so on

`Linux arch 6.4.11-arch2-1 #1 SMP PREEMPT_DYNAMIC Sat, 19 Aug 2023 15:38:34 +0000 x86_64 GNU/Linux`

### Description

Run the following command to eat 100% of memory:
```
identify -format '%[EXIF:Orientation]\n' --  file.djvu
```

Related: https://github.com/ranger/ranger/issues/1815

### Steps to Reproduce

Here's the Dockerfile:
```Dockerfile
FROM archlinux:base-devel

RUN pacman -Syu --noconfirm

RUN pacman -S --noconfirm djvulibre

RUN mkdir /magick
WORKDIR /magick
COPY . .

RUN ./configure
RUN make install

ADD https://filesamples.com/samples/document/djvu/sample2.djvu /test/
```

```sh
$ git clone https://github.com/ImageMagick/ImageMagick
$ cd ImageMagick
$ git checkout 2b42758
# copy above docker file, save as Dockerfile. 
$ docker build -t repro6584 .
$ docker run -it repro6584
# in the shell insider docker:

# djvulibre works with no problem
$ ddjvu -format=tiff -quality=90 -page=1 -size=1000x1000 - /tmp/a < /test/sample2.djvu

# this will consume all memory
$ identify -format '%[EXIF:Orientation]\n' --  /test/sample2.djvu
```

**Comment by dlemstra** [comment]
Can you please create a Dockerfile that can be used to reproduce this issue?

**Comment by glyh** [comment]
BTW this should have a high priority as ranger use identify to kill my computer at least 30 times.

**Comment by dlemstra** [comment]
You don't get to decide the priority of our work. We work on this project in our spare time so we get to decide when to work on what. 

And if you provide us with a `Dockerfile` you will be saving us a lot of time. Otherwise we would need to figure out what combination of software and operating system is causing this issue. Have you tried different versions of the `djvulibre` library? This could also be a problem in their library. But it might also be that our software is not compatible with a specific version of the `djvulibre` library.

**Comment by glyh** [comment]
djvulibre is working correctly, here's a snippet how ranger worked around it when identifying the image is needed:
```sh
    case "${mimetype}" in

        ## DjVu
        image/vnd.djvu)
            ddjvu -format=tiff -quality=90 -page=1 -size="${DEFAULT_SIZE}" \
                  - "${IMAGE_CACHE_PATH}" < "${FILE_PATH}" \
                  && exit 6 || exit 1;;

        ## Image
        image/*)
            local orientation
            orientation="$( identify -format '%[EXIF:Orientation]\n' -- "${FILE_PATH}" )"
            ## If orientation data is present and the image actually
            ## needs rotating ("1" means no rotation)...
            if [[ -n "$orientation" && "$orientation" != 1 ]]; then
                ## ...auto-rotate the image according to the EXIF data.
                convert -- "${FILE_PATH}" -auto-orient "${IMAGE_CACHE_PATH}" && exit 6
            fi

            ## `w3mimgdisplay` will be called for all images (unless overridden
            ## as above), but might fail for unsupported types.
            exit 7;;

```

**Comment by glyh** [comment]
I updated the issue with the dockerfile, you can test it out.

**Comment by urban-warrior** [comment]
Download the latest from Github main and use this command:

```
identify -format '%[orientation]\n' 'sample2.djvu[0]'
```

Lightening fast with minimal memory footprint.

**Comment by glyh** [comment]
I just test it out it's still consuming my memory, I'll try again inside of docker to confirm it.

**Comment by urban-warrior** [comment]
Do you download https://github.com/ImageMagick/ImageMagick/commit/b114ea41095707bda4da852900abecc109920bc1?   Prior to the patch, ImageMagick consumed 7GB of memory and ran in  14 seconds.  With the patch, it consumes 0.5MB of memory and completes in 0.22 seconds.

**Comment by glyh** [comment]
Sorry I didn't operate correctly, let me try again on AUR's git package.

**Comment by glyh** [comment]
Tested on my end on my local system rather than docker, 4804d6124 is still buggy, did you tested it inside docker?

**Comment by urban-warrior** [comment]
https://github.com/ImageMagick/ImageMagick/commit/4804d612486db0be8941b89ce93890e9d6eb61c7 is not the patch you want.  You want https://github.com/ImageMagick/ImageMagick/commit/b114ea41095707bda4da852900abecc109920bc1.

**Comment by glyh** [comment]
4804d61 is older than that, so it must mean either the fix isn't valid or the later commit brokes it again. I'll test that out.

**Comment by glyh** [comment]
https://github.com/ImageMagick/ImageMagick/commit/b114ea41095707bda4da852900abecc109920bc1 is older than that, so it must mean either the fix isn't valid or the later commit brokes it again. I'll test that out. 

EDIT: 
- https://github.com/ImageMagick/ImageMagick/commit/b114ea41095707bda4da852900abecc109920bc1 is working in my docker container, let me test it on my machine. 
- https://github.com/ImageMagick/ImageMagick/commit/b114ea41095707bda4da852900abecc109920bc1 is not working on my machine, I'm at this point no clue how should I reproduce it in docker.

**Comment by glyh** [comment]
Here's my configure.ac if anyone is interested to reproduce, you may need to install a bunch of related packages tho. 
https://pastebin.ubuntu.com/p/gwHsvCPm3x/

**Comment by dlemstra** [comment]
Closing this issue because we have a applied a change that improves loading times when only metadata is required.

**Comment by glyh** [comment]
Well technically it works better than before on my end, now it consumes only 70% of RAM and take 30s to finish the task. So this is technically fixed.
